core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -264,6 +264,14 @@ public static boolean isSameColumnAndDifferentDatatypeInSchema(
             .equalsIgnoreCase(tableColumnList.get(i).getColumnName()) && !indexFileColumnList.get(j)
             .getDataType().getName()
             .equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+          if ("varchar".equalsIgnoreCase(indexFileColumnList.get(j).getDataType().getName()) &&
+              "string".equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+            LOG.info("Datatype of the Column " + indexFileColumnList.get(j).getDataType().getName()
+                + " present in index file, is varchar and not same as datatype of the column " +
+                "with same name present in table, " +
+                "because carbon convert varchar of carbon to string of spark");
+            return true;
```

Comment:
I think here you should not return , should continue with the loop as the for loop is not yet completed for all columns and other column might have different datatype.

Reply:
ok,done